BT_CTLR_SW_SWITCH_SINGLE_TIMER cannot be used with nrf53 targets

**Describe the bug**
BT_CTLR_SW_SWITCH_SINGLE_TIMER  kconfig indicates this option can be used with an nrf5340
but it cannot be built for those targets.

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf5340dk_nrf5340_cpunet ../samples/bluetooth/peripheral_hr/ -DCONFIG_BT_CTLR_ADVANCED_FEATURES=y -DCONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER=y
3. ninja
4. See error

**Expected behavior**
Either kconfig prevents the selection, or it builds properly

**Impact**
Confused developers.

**Logs and console output**
```
subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_resources.h:30:32: error: 'NRF_TIMER4' undeclared (first use in this function); did you mean 'NRF_TIMER0'?
   30 | #define EVENT_TIMER    _CONCAT(NRF_TIMER, EVENT_TIMER_ID)
      |                                ^~~~~~~~~
include/zephyr/toolchain/common.h:137:26: note: in definition of macro '_DO_CONCAT'
  137 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_resources.h:30:24: note: in expansion of macro '_CONCAT'
   30 | #define EVENT_TIMER    _CONCAT(NRF_TIMER, EVENT_TIMER_ID)
      |                        ^~~~~~~
subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi.h:25:39: note: in expansion of macro 'EVENT_TIMER'
   25 |                 nrf_timer_publish_set(EVENT_TIMER, NRF_TIMER_EVENT_COMPARE0,
      |                                       ^~~~~~~~~~~
[..]
```

**Environment (please complete the following information):**
 - OS: Unbuntu 22.04
 - Toolchain: Zephyr SDK 0.16.1
 - Commit SHA or Version used: today's main 6267259f91d2561666a97dd836e4ea7e2aabbc3a

--- a/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5340.h
+++ b/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5340.h
@@ -5,6 +5,20 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/* Override EVENT_TIMER_ID from 4 to 0, as nRF5340 does not have 4 timer
+ * instances.
+ */
+#if defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+#undef EVENT_TIMER_ID
+#define EVENT_TIMER_ID 0
+
+#undef EVENT_TIMER
+#define EVENT_TIMER    _CONCAT(NRF_TIMER, EVENT_TIMER_ID)
+
+#undef SW_SWITCH_TIMER
+#define SW_SWITCH_TIMER EVENT_TIMER
+#endif /* CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER */
+
 /* NRF Radio HW timing constants
  * - provided in US and NS (for higher granularity)
  * - based on empirical measurements and sniffer logs

--- a/tests/bluetooth/init/testcase.yaml
+++ b/tests/bluetooth/init/testcase.yaml
@@ -122,6 +122,19 @@ tests:
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf52dk_nrf52832
+  bluetooth.init.test_ctlr_sw_switch_single_timer:
+    extra_args:
+      - CONF_FILE=prj_ctlr.conf
+      - CONFIG_BT_CTLR_ADVANCED_FEATURES=y
+      - CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER=y
+    platform_allow:
+      - nrf5340dk_nrf5340_cpunet
+      - nrf52840dk_nrf52840
+      - nrf52dk_nrf52832
+    integration_platforms:
+      - nrf5340dk_nrf5340_cpunet
+      - nrf52840dk_nrf52840
+      - nrf52dk_nrf52832
   bluetooth.init.test_ctlr_ticker:
     extra_args:
       - CONF_FILE=prj_ctlr_ticker.conf

